File: CS/CustomSavingEUD/Form1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraEditors;
using DevExpress.XtraReports.Design;
using DevExpress.XtraReports.Parameters;
using DevExpress.XtraReports.UserDesigner;

namespace CustomSavingEUD {
    public partial class Form1 : DevExpress.XtraEditors.XtraForm {
        public Form1() {
            InitializeComponent();
        }
        private void button1_Click(object sender, EventArgs e) {
            ShowReportDesigner();
        }
        private void ShowReportDesigner() {
            XRDesignForm form = new XRDesignForm();
            XRDesignMdiController mdiController = form.DesignMdiController;
            mdiController.DesignPanelLoaded += new DesignerLoadedEventHandler(mdiController_DesignPanelLoaded);
            mdiController.OpenReport(new XtraReport1());
            form.ShowDialog();
            if (mdiController.ActiveDesignPanel != null) {
                mdiController.ActiveDesignPanel.CloseReport();
            }
        }
        private void mdiController_DesignPanelLoaded(object sender, DesignerLoadedEventArgs e) {
            ReportTabControl tabControl = e.DesignerHost.GetService(typeof(ReportTabControl)) as ReportTabControl;
            if (tabControl == null) {
                return;
            }
            tabControl.PreviewReportCreated += tabControl_PreviewReportCreated;
        }
        private void tabControl_PreviewReportCreated(object sender, EventArgs e) {
            ReportTabControl tabControl = sender as ReportTabControl;
            if (tabControl != null && tabControl.PreviewReport != null) {
                tabControl.PreviewReport.ParametersRequestBeforeShow += PreviewReport_ParametersRequestBeforeShow;
            }
        }
        private void PreviewReport_ParametersRequestBeforeShow(object sender, ParametersRequestEventArgs e) {
            //customize parameters
            if (e.ParametersInformation.Length == 0) {
                return;
            }
            ParameterInfo info = e.ParametersInformation[0];
            if (info.Parameter.Type == typeof(Int32)) {
                info.Editor = new SpinEdit();
                info.Parameter.Value = 100;
            }
        }
    }
}

File: VB/CustomSavingEUD/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraEditors
Imports DevExpress.XtraReports.Design
Imports DevExpress.XtraReports.Parameters
Imports DevExpress.XtraReports.UserDesigner

Namespace CustomSavingEUD
    Partial Public Class Form1
        Inherits DevExpress.XtraEditors.XtraForm

        Public Sub New()
            InitializeComponent()
        End Sub
        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles simpleButton1.Click
            ShowReportDesigner()
        End Sub
        Private Sub ShowReportDesigner()
            Dim form As New XRDesignForm()
            Dim mdiController As XRDesignMdiController = form.DesignMdiController
            AddHandler mdiController.DesignPanelLoaded, AddressOf mdiController_DesignPanelLoaded
            mdiController.OpenReport(New XtraReport1())
            form.ShowDialog()
            If mdiController.ActiveDesignPanel IsNot Nothing Then
                mdiController.ActiveDesignPanel.CloseReport()
            End If
        End Sub
        Private Sub mdiController_DesignPanelLoaded(ByVal sender As Object, ByVal e As DesignerLoadedEventArgs)
            Dim tabControl As ReportTabControl = TryCast(e.DesignerHost.GetService(GetType(ReportTabControl)), ReportTabControl)
            If tabControl Is Nothing Then
                Return
            End If
            AddHandler tabControl.PreviewReportCreated, AddressOf tabControl_PreviewReportCreated
        End Sub
        Private Sub tabControl_PreviewReportCreated(ByVal sender As Object, ByVal e As EventArgs)
            Dim tabControl As ReportTabControl = TryCast(sender, ReportTabControl)
            If tabControl IsNot Nothing AndAlso tabControl.PreviewReport IsNot Nothing Then
                AddHandler tabControl.PreviewReport.ParametersRequestBeforeShow, AddressOf PreviewReport_ParametersRequestBeforeShow
            End If
        End Sub
        Private Sub PreviewReport_ParametersRequestBeforeShow(ByVal sender As Object, ByVal e As ParametersRequestEventArgs)
            'customize parameters
            If e.ParametersInformation.Length = 0 Then
                Return
            End If
            Dim info As ParameterInfo = e.ParametersInformation(0)
            If info.Parameter.Type Is GetType(Int32) Then
                info.Editor = New SpinEdit()
                info.Parameter.Value = 100
            End If
        End Sub
    End Class
End Namespace
